tools/wavetool.py

```diff
@@ -0,0 +1,146 @@
+#! /usr/bin/python3
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright(c) 2020 Intel Corporation. All rights reserved.
+
+"""
+A Tool to Generate and Manipulate Wave Files
+"""
+
+import os
+import sys
+import argparse
+import numpy as np
+import scipy.signal as signal
+import scipy.io.wavfile as wavefile
+
+eps = 0.0000000001 # used to avoid zero division error
+
+def generate_sinusoids(**p):
+    """
+    Generate sine or cosine wave.
+
+    ``y(t) = A * sin(2 * pi * f * t + phi)``
+
+    ``y(t) = A * cos(2 * pi * f * t + phi)``
+
+    Parameters
+    ----------
+    dict of sinusiod parameters:
+    p['type']: Wave type, ``sine`` or ``cosine``
+    p['amp']: Amplitude
+    p['freq']: Frequency
+    p['phase']: Phase
+    p['chan']: Channels
+    p['sample_rate']: Sample rate
+    p['duration']: Duration
+
+    Returns
+    ----------
+    Real n-D Array with shape ``(duration * sample_rate, channel)``
+    """
+    func = np.sin if p['type'] == 'sine' else np.cos
+    time = np.arange(0, p['duration'], 1.0 / p['sample_rate'], dtype=np.float32)
+    data = p['amp'] * func(2 * np.pi * p['freq'] * time + p['phase'])
+    return np.reshape(np.repeat(data, p['chan']),[len(data), p['chan']])
+
+def generate_wav(**cmd):
+    if cmd['generate'] in ['sine', 'cosine']:
+        wave_param = {
+            'type': cmd['generate'], 'amp': cmd['amp'][0], 'freq': cmd['freq'][0],
+            'phase': cmd['phase'][0], 'chan': cmd['channel'], 'sample_rate': cmd['sample_rate'],
+            'duration': cmd['duration'][0]
+        }
+        wave_data = generate_sinusoids(**wave_param)
+        return wave_data
```

Comment:
It looks like you want `generate_wav()` to be ready for other types of signals, correct? If correct then make it completely ready and clearer by:

1. adding the missing `default case`
```python
    else:
        raise Exception("invalid generate function")
```
... you never want this function to silently return `None` in case of a future bug.

2. Indenting `return wave_data` one notch left